### PR TITLE
[FIX] l10n_fr: support french fiscal rounding

### PR DESCRIPTION
**Pull request by JulienVR**
Each line of the french VAT report should be rounded to the unit. In addition, the tax amounts should be recomputed using the rounded base amounts!

The rounding of each line is enabled using the 'integer_rounding' option introduced in
https://github.com/odoo/enterprise/commit/1916433a951fb98baf1a54e6b6e0f8a3f59920a4. The recomputation of the tax lines base on the rounded base lines is done using aggregation expression (so the `tax` tags are no longer used in the report).

The rounding difference is accounted during the closing entry.

task-3607253

### CODE REVIEW

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_fr/data/tax_report_data.xml [hunk 1]
```diff
@@ -1356,7 +1451,6 @@
                                 <field name="engine">aggregation</field>
                                 <field name="formula">box_27.balance</field>
                                 <field name="carryover_target">box_22._applied_carryover_balance</field>
-                                <field name="subformula">round(0)</field>
```
[the remark this answers is not in the document]
Comment (JulienVR) [reply addons/l10n_fr/data/tax_report_data.xml:1359]: I set it to `''` to make sure we match the first condition in `_aggregation_apply_bounds` and return directly :)

[the remark this answers is not in the document]
Comment (JulienVR) [reply addons/l10n_fr/data/tax_report_data.xml:1359]: nvm I'm dumb (set it to `eval='False'`)

**Review by oco-odoo — APPROVED** [review]
@robodoo r+

**Comment by oco-odoo** [comment]
@robodoo r-

Duplicate xmlids

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Comment by Levizar** [comment]
@robodoo rebase-ff

**Pull request closed without merging**